redis/asyncio/connection.py

```diff
@@ -718,7 +720,14 @@ async def on_connect(self) -> None:
             await self.send_command("CLIENT", "SETNAME", self.client_name)
             if str_if_bytes(await self.read_response()) != "OK":
                 raise ConnectionError("Error setting client name")
-
+        if self.is_connected is False:
```

Comment:
It's for the test_cwe_404.py in test_asyncio.In the test_standalone_pipeline test case, an error will be raised if the statement is not added when single_connection_client is true.

You can try running this test case.

```python
@pytest.mark.onlynoncluster
@pytest.mark.parametrize("delay", argvalues=[0.05, 0.5, 1, 2])
async def test_standalone_pipeline(delay):
    dp = DelayProxy(
        addr=("localhost", 5380), redis_addr=("localhost", 6379), delay=delay * 2
    )
    await dp.start()
    for b in [True, False]:
        async with Redis(host="localhost", port=5380, single_connection_client=b) as r:
            await r.set("foo", "foo")
            await r.set("bar", "bar")

            pipe = r.pipeline()

            pipe2 = r.pipeline()
            pipe2.get("bar")
            pipe2.ping()
            pipe2.get("foo")

            t = asyncio.create_task(pipe.get("foo").execute())
            await asyncio.sleep(delay)
            t.cancel()

            pipe.get("bar")
            pipe.ping()
            pipe.get("foo")
            pipe.reset()

            assert await pipe.execute() is None

            # validating that the pipeline can be used as it could previously
            pipe.get("bar")
            pipe.ping()
            pipe.get("foo")
            assert await pipe.execute() == [b"bar", True, b"foo"]
            assert await pipe2.execute() == [b"bar", True, b"foo"]

    await dp.stop()

Reply:
Can you add test that check the on_connect() set library name & version without calling client_setinfo() ? I think it won't work with this if condition...